src/applications/caregivers/config/submit-transformer.js

```diff
@@ -11,6 +11,12 @@ const submitTransformer = (formConfig, form) => {
     ? 'secondaryTwo'
     : null;
 
+  if (formData['view:useFacilitiesAPI']) {
```

Comment:
If the toggle is on we set the `veteranPlannedClinic` from the `view:plannedClinicId` object. This ensures we are passing the facility id in the correct format and the properly selected one.

Reply:
Good callout. I do not know. This is one where we probably _do_ want to turn it on for a limited subset of users, so we should probably look into this.